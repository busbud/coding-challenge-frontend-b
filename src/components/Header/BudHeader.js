import React from 'react';

export default function BudHeader(props){
    return(
        <div className="header">
            <h1>Welcome to Osheaga 2020 event!<br></br>Please, take a seat!</h1>
            <h4>We are really pleased to help you finding your best ticket for the Osheaga festival 2020. Please, keep safe and have fun!</h4>
        </div>
    )
}