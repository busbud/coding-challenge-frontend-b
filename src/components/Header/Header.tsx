import React from 'react';
import styled from 'styled-components';
import { Container } from '@material-ui/core';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { changeLang, langSelector, Lang } from '../../store/lang';
import { AppState } from '../../store';
import Button from '../Button';

interface Props {
  onClickAction: (payload: Lang) => void;
  lang: Lang;
}

const StyledHeader = styled.header`
  align-items: center;
  background-color: ${props => props.theme.busbud.header.backgroundColor};
  color: ${props => props.theme.busbud.header.color};
  display: flex;
  padding: ${props => props.theme.busbud.header.padding};
`;

const StyledContainer = styled(Container)`
  align-items: center;
  display: flex;
  justify-content: space-between;
`;

const UnordoredList = styled.ul``;
const ListItem = styled.li`
  display: inline;
  margin-left: 10px;
`;

function Header({ onClickAction }: Props) {
  const handleChangeLang = (newLang: Lang) => {
    onClickAction(newLang);
  };
  return (
    <StyledHeader>
      <StyledContainer>
        <FormattedMessage id="title" />
        <UnordoredList>
          <ListItem>
            <Button onClickAction={() => handleChangeLang('en')}>
              <FormattedMessage id="english" />
            </Button>
          </ListItem>
          <ListItem>
            <Button onClickAction={() => handleChangeLang('fr')}>
              <FormattedMessage id="french" />
            </Button>
          </ListItem>
        </UnordoredList>
      </StyledContainer>
    </StyledHeader>
  );
}

const mapStateToProps = (state: AppState) => ({
  lang: langSelector(state)
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  onClickAction: (lang: Lang) => dispatch(changeLang(lang))
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
