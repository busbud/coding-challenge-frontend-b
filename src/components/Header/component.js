import React from 'react';
import { object } from 'prop-types';
import AppBar from 'material-ui/AppBar';
import Grid from 'material-ui/Grid';
import Toolbar from 'material-ui/Toolbar';
import Logo from '../Logo';
import LocalesSelector from '../../containers/LocalesSelector';

const Component = ({ classes }) => (
  <AppBar className={classes.root} position="sticky" color="secondary">
    <Toolbar className={classes.toolbar}>
      <Grid container justify="center" className={classes.root}>
        <Grid item md={8} xs={11} className={classes.item}>
          <Logo />
          <LocalesSelector />
        </Grid>
      </Grid>
    </Toolbar>
  </AppBar>
);

Component.propTypes = {
  classes: object,
};

export default Component;
