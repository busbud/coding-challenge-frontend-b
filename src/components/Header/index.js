import React from 'react'
import { connect } from 'react-redux'
import moment from 'moment'

import { ArrowRight } from 'styled-icons/feather'

import * as schedulesSelectors from '../../store/services/schedules/selectors'

import Container from '../Container'

import {
  StyledContent,
  StyledDate,
  StyledHeader,
  StyledTitle,
} from './styles'

const Header = (props) => (
  <StyledHeader>
    <Container>
      <StyledContent>
        <StyledTitle>
          {props.cities[0] && props.cities[0].name}
          <ArrowRight size={26} />
          {props.cities[1] && props.cities[1].name}
        </StyledTitle>

        <StyledDate>
          {moment(props.date).format('LL')}
        </StyledDate>
      </StyledContent>
    </Container>
  </StyledHeader>
)

const mapStateToProps = state => ({
  cities: schedulesSelectors.getCities(state),
})

export default connect(mapStateToProps)(Header)