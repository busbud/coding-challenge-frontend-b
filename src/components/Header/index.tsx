import * as React from 'react'
import LangSwitcher from '../../components/LangSwitcher'
import CurrencySwitcher from '../../components/CurrencySwitcher'

interface TopBarProps {
  onLangChange: (ev: React.MouseEvent<HTMLButtonElement>) => void
  onCurrencyChange: (ev: React.MouseEvent<HTMLButtonElement>) => void
}

export default class TopBar extends React.Component<TopBarProps, any> {
  constructor (props: TopBarProps) {
    super(props)
  }

  render () {
    return (
      <div className='header-top-bar'>
        <div className='container'>
          <div className='row align-items-center'>
            <div className='col-md-3 header-top-bar-logo'></div>
            <div className='col-md-9 header-top-bar-lang'>
              <LangSwitcher langChange={this.props.onLangChange}/>
              <CurrencySwitcher currencyChange={this.props.onCurrencyChange}/>
            </div>
          </div>
        </div>
      </div>
    )
  }
}
