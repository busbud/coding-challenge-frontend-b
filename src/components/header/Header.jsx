import React from "react";

export const Header = () => <div>header</div>;
