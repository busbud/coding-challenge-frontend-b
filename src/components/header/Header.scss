.lang {
    display: flex;
    justify-content: flex-end;
    background-color: #287FC6;

    div {
        padding: 0 10px;
        cursor: pointer;
    }
}