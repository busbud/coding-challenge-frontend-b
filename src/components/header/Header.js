import React from 'react';
import PropTypes from 'prop-types';
import './Header.scss';
import withI18n from '../../contexts/withI18n';

const Header = props => {

    const handleChangeLanguage = lang => props.i18n.changeLanguage(lang);

    return (
        <div className="lang">
            <div className="lang">
                <div onClick={() => handleChangeLanguage('en')}>ğŸ‡¬ğŸ‡§</div>
                <div onClick={() => handleChangeLanguage('fr')}>ğŸ‡«ğŸ‡·</div>
            </div>
        </div>
    )
}

Header.propTypes = {
    i18n: PropTypes.shape({
        changeLanguage: PropTypes.func
    }).isRequired
}

export const header = Header;
export default withI18n(Header);