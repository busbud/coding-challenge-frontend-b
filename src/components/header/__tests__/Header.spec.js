import React from 'react';
import renderer from 'react-test-renderer';
import {header as Header} from '../Header';

const props = {
    i18n: {
        changeLanguage: () => {}
    }
}

describe('Header', () => {
    it('renders correctly', () => {
        const tree = renderer
          .create(<Header {...props} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
})
