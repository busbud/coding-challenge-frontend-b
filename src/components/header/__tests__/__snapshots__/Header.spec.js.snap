// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="lang"
>
  <div
    className="lang"
  >
    <div
      onClick={[Function]}
    >
      ğŸ‡¬ğŸ‡§
    </div>
    <div
      onClick={[Function]}
    >
      ğŸ‡«ğŸ‡·
    </div>
  </div>
</div>
`;
